prusti-tests/tests/parse/fail/ghost-constraints/trait-bounds-illegal-2.rs

```diff
@@ -2,9 +2,10 @@ use prusti_contracts::*;
 
 trait A<X> { }
 
-#[ghost_constraint(T: for<'a> A<&'a i32> , [ //~ ERROR: lifetimes are not allowed in ghost constraint trait bounds
+#[ghost_constraint(T: for<'a> A<&'a i32>, [ //~ ERROR: lifetimes are not allowed in ghost constraint trait bounds
     ensures(result > 0)
 ])]
+//~| ERROR: expected a trait bound `T: A + B` and specifications in brackets `[requires(...), ensures(...), pure, ...]`
```

Comment:
Here the extra error is maybe unnecessary, but we would have to check (by string comparison?) that the emitted error is not just a generic `syn` parse error, right?

Reply:
I have to specify all errors that the test expects to generate; any additional or missing ones will cause it to fail. The way syn outputs these combined errors is as two separate errors with the same span (closest we can get to a note).

Reply:
I understand that but here our error is specifically that we disallow lifetimes, which is an error we emit ourselves, as opposed to a generic `syn` parsing error.

Reply:
Ah, my bad, I see what you mean now. Yeah, it's kind of silly to attach an error like this in that case. It doesn't do much harm, so I wonder if it justifies the additional complexity of moving down the error to more specific parsing logic—I'll push a commit that does that so we can compare.